src/openassetio-core/include/openassetio/macros.hpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2025 The Foundry Visionmongers Ltd
+#pragma once
+/**
+ * Forward declare a shared_ptr for a given class.
+ *
+ * Assumes the surrounding namespace is valid for the given class.
+ *
+ * Declares both non-const and const variants as `ClassPtr` and
+ * `ClassConstPtr`, respectively.
+ */
+#define OPENASSETIO_DECLARE_PTR(Class)             \
+  /* NOLINTNEXTLINE(bugprone-macro-parentheses) */ \
+  class Class;                                     \
+  using Class##Ptr = std::shared_ptr<Class>;       \ // NOLINT(build/include_what_you_use)
```

Comment:
[fix] Looks like CI is failing because the comment comes after the `\`.  I think you'll have to put the comment in `/* ... */` before the `\`.

Reply:
I'm kinda surprised this works, i.e. that the NOLINT applies to the following line too, but cpplint isn't complaining about this specific issue any more.

However, now both cpplint and clang-tidy [is complaining](https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/15619139028/job/44036846770?pr=1461), I think because there needs to be two spaces before the `// NOLINT`.

Again, it would save a lot of time if you could build locally to check the linters are happy before pushing.